modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Form/ImportCsvForm.php

```diff
@@ -108,10 +113,37 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $migration_group_id = $form_state->getValue('migration_group');
-    $count = (int) $form_state->getValue('count');
-    $migrations = $this->pluginManager->createInstances([]);
-    ImportCsvBatch::buildBatch($migrations, $migration_group_id, $count);
+    $migration = $this->entityTypeManager
+      ->getStorage('migration')
+      ->load('gc_auth_custom_users');
+    if (!$migration) {
+      $form_state->setErrorByName('csv_file', $this->t('Migration "gc_auth_custom_users" with file destination settings not found.'));
+    }
+    $file_path = $migration->get('source')['path'];
+    $dir = substr($file_path, 0, strrpos($file_path, '/', -1));
+    $url = Url::fromRoute('openy_gc_auth.provider.edit', ['type' => 'custom']);
+    $form_state->setRedirectUrl($url);
+
+    $fid = $form_state->getValue(['csv_file', 0]);
+    $csv_file_id = $form_state->get('csv_file_id');
+    if ($fid == $csv_file_id) {
+      return;
+    }
+
+    $fid = $form_state->getValue(['csv_file', 0]);
+    $csv_file_id = $form_state->get('csv_file_id');
```

Comment:
The same assignments are done a few lines above.

Reply:
fixed